source/tomopy/misc/corr.py

```diff
@@ -267,6 +268,168 @@ def median_filter_cuda(arr, size=3, axis=0):
+    ----------
+    x_idx : int
+        The x index for a given nonfinite value.
+    y_idx :  int
+        The y index for a given nonfinite value.
+    shape_x : int
+        The max shape of the 2D projection in the x dimension.
+    shape_y : int
+        The max shape of the 2D projection in the y dimension.
+
+    Returns
+    -------
+    ndarray
+        The integer kernel bounds to surround the nonfinite value with.
+    """
+
+    # Determining x kernel
+    x_idx_lower = x_idx - kernel
+    x_idx_higher = x_idx + kernel + 1
+
+    if x_idx_lower < 0:
+        x_idx_lower = 0
+    if x_idx_higher > shape_x-1:
+        x_idx_higher = shape_x-1
+
+    # Determining y kernel
+    y_idx_lower = y_idx - kernel
+    y_idx_higher = y_idx + kernel + 1
+
+    if y_idx_lower < 0:
+        y_idx_lower = 0
+    if y_idx_higher > shape_y-1:
+        y_idx_higher = shape_y-1
+
+    # Forcing values to be integers
+    integer_values_pre = [x_idx_lower, x_idx_higher, y_idx_lower, y_idx_higher]
+    integer_values = [int(i) for i in integer_values_pre]
+
+    return integer_values
+
+
+def median_filter_nonfinite(data, kernel=1, callback=None):
+    """
+    Apply a selective 2D median filter, slice by slice, for a 3D data array
+    with filter_size=kernel to all nonfinite values in the 3D data array.
+
+    Parameters
+    ----------
+    data : ndarray
+        The 3D array of data with nonfinite values in it.
+    kernel : int
+        The size of the kernel to be used for a local median filter. 
+    callback : func
+        A callback function to be run for each loop iteration.
+        Use self.iterator_len and self.iterator_desc to obtain the
+        length and description for each loop
+
+    Returns
+    -------
+    ndarray
+        The corrected 3D array with all nonfinite values removed based upon the local
+        median value defined by the kernel size.
+    """
+
+    # Defining a callback class for a User defined progressbar
+    class CallbackClass():
+
+        def __init__(self, iterator_len, iterator_desc):
+            self.iterator_len = iterator_len
+            self.iterator_desc = iterator_desc
+
+    # Creating store arrays for the prj, x, y indicies for bad values
+    nonfinite_idx_store = [[], [], []]
+
+    # Allowing the User to use a progressbar if desired
+    if callback is not None:
+        data_len_loop1 = len(data)
+        callback_loop1 = CallbackClass(
+            iterator_len=data_len_loop1, iterator_desc='Finding Bad Values')
+        callback_loop1.callback = callback
+
+    # Iterating throug each projection to save on RAM
+    for idx, projection in enumerate(data):
+        nonfinite_idx_store = np.concatenate((nonfinite_idx_store,
+                                              _find_nonfinite_values_prj(projection,
+                                                                         int(idx))), axis=1)
+        if callback is not None:
+            callback_loop1.callback(callback_loop1)
```

Comment:
Instead of checking if the callback is None every iteration, always call the callback. If the user doesn't provide a callback, instantiate a function that does nothing:
```python
# This goes at the top
if callback is None:
    def callback(...):
        pass

for x in y:
    # do stuff
     callback(...)
    # the callback is always called regardless of whether the user provided it
```